### DISCUSSION TITLE
struct vs readonly struct vs ref struct vs record struct

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#8] [General]
# struct vs readonly struct vs ref struct vs record struct

C# knows struct since its down of time. But there are also recent additions like readonly struct, record struct  and ref struct.
This article will show what are the differences between those 4.

https://steven-giesel.com/blogPost/2a52cd8d-b3b2-42e4-87e8-d6dc14147ddb

**Comment by ghost** [comment]
Thanks for writing this, it really clarified the differences between the various struct types.

**Reply by linkdotnet** [reply]
Thanks for the kind words!

**Comment by haveve** [comment]
Sharing my thanks to you. It is good post for all who wanna know a little depply about stuct.
